splinepy/helpme/create.py

```diff
@@ -298,6 +298,306 @@ def revolved(
+            curv.append(
+                round(_np.linalg.norm(trajectory.derivative([i], [2])), 2)
+            )
+        # evaluate the par_values-vector indices of the maximum curvature points
+        max_curv = max(curv)
+        max_indices = [i for i, x in enumerate(curv) if x == max_curv]
+        # prepare matrix for the insertion
+        insertion_values = []
+        # compute the new insertion values
+        par_values = par_value.ravel()
+
+        for maxi in max_indices:
+            if maxi == 0:
+                insertion_values.append(
+                    (par_values[maxi] + par_values[maxi + 1]) / 2
+                )
+            elif maxi == len(par_values) - 1:
+                insertion_values.append(
+                    (par_values[maxi] + par_values[maxi - 1]) / 2
+                )
+            else:
+                insertion_values.append(
+                    (par_values[maxi] + par_values[maxi - 1]) / 2
+                )
+                insertion_values.append(
+                    (par_values[maxi] + par_values[maxi + 1]) / 2
+                )
+
+        # insert knots into the trajectory's knot vector
+        insertion_values = _np.unique(insertion_values)
+        # convert knot vector to list
+        kv_list = trajectory.knot_vectors[0].numpy()
+
+        # insert knots into the trajectory's knot vector
+        if any(value in insertion_values for value in kv_list):
+            add = _np.concatenate((insertion_values, kv_list))
+            add = _np.unique(add)
+            # remove the existing knots
+            add = add[~_np.isin(add, kv_list)]
+            # check if there are any knots to insert
+            if len(add) == 0:
+                _log.warning("Auto Refinement couldn't insert knots.")
+            else:
+                trajectory.insert_knots(0, add)
+                # give information about the inserted knots
+                _log.info(
+                    f"Auto Refinement inserted {len(add)} "
+                    "knots into the trajectory."
+                )
+        else:
+            trajectory.insert_knots(0, insertion_values)
+            # give information about the inserted knots
+            _log.info(
+                f"Auto Refinement inserted {len(insertion_values)} "
+                "knots into the trajectory."
+            )
+
+        # recalculate parameter values
+        par_value = trajectory.greville_abscissae()
+        par_value = par_value.reshape(-1, 1)
+
+    ### SWEEPING PROCESS ###
+
+    # evaluate center of cross section and translate to origin
+    cross_para_center = _np.mean(cross_section.parametric_bounds, axis=0)
+    cs_center = cross_section.evaluate(
+        cross_para_center.reshape(-1, cross_section.para_dim)
+    ).ravel()
+    cross_section.control_points -= cs_center
+
+    # evaluate transformation matrices for every trajectory point
+    A, R = transformation_matrices(trajectory, par_value)
```

Comment:
you are using this transformation_matrice just once - why do you create a local function? or can you generalize and put it somewhere else?

Reply:
I made a function because I called it in a for-loop previously. Since I don't do this anymore, using `transformation_matrices `as a function isn't necessary anymore, you're right. 
I've removed the function and it's a "normal" part of the code now.